examples/fedpca_examples/perform_pca/README.md

```diff
@@ -0,0 +1,41 @@
+# Federated Principal Component Analysis Example
+This example performs federated principal component analysis. The goal is to compute the principal components of a subset of the MNIST dataset, under the condition that the data is distributed across two distinct clients.
+
+This is achieved by each client performing PCA locally at first, then the principal components are sent to a central server to be merged.
+
+Each client performs Dirichlet subsampling on the whole dataset to produce heterogeneous local datasets. This is done to ensure that local principal compoents are distinct enough across different clients.
+
+## Running the Example
+In order to run the example, first ensure you have [installed the dependencies in your virtual environment according to the main README](/README.md#development-requirements) and it has been activated.
+
+## Starting Server
+
+The next step is to start the server by running
+```
+python -m examples.fedpca_examples.perform_pca.server  --config_path /path/to/config.yaml
+```
+from the FL4Health directory. The following arguments must be present in the specified config file:
+* `n_clients`: number of clients the server waits for in order to run the FL training
+* `batch_size`: size of the batches each client will train on
+* `low_rank`: whether the data matrix is low rank or not. If the user has prior knowledge that the data matrix is low rank, then this parameter can be set to True so that each client can leverage this knowledge to perform local PCA more efficiently.
+* `full_svd`: determines whether full SVD or reduced SVD is performed by each client.
+* `rank_estimation`: an estimation of the rank of the data matrix. This is only used if `low_rank` is set to True.
+* `center_data`: if set to True, the mean of data will be subtracted from all data points before local PCA is performed.
+* `num_components`: by default, after merging is completed, each client will use the merged principal components to compute the reconstruction error on its validation set. This parameter specify the number of principal components that are used in this evaluation process.
+
+## Starting Clients
+
+Once the server has started and logged "FL starting," the next step, in separate terminals, is to start the two
+clients. This is done by simply running (remembering to activate your environment)
+```
+python -m examples.fedpca_examples.perform_pca.client --dataset_path /path/to/data --model_save_path /path/to/save/model
```

Comment:
It might be better to name with something like `principal_components_save_path` or just `components_save_path` to avoid confusion with a full ML model? There might be a reason that I'm not foreseeing to not use these names though.

Reply:
Yes I agree that would be clearer. I originally named it `model_save_path` because the PCA module is implemented as a PyTorch "model"